test/e2e/app_management_test.go

```diff
@@ -547,6 +548,60 @@ func TestResourceDiffing(t *testing.T) {
 			diffOutput, err := RunCli("app", "diff", app.Name, "--local", "testdata/guestbook")
 			assert.NoError(t, err)
 			assert.Empty(t, diffOutput)
+		}).
+		Given().
+		When().
+		And(func() {
+			output, err := RunWithStdin(testdata.SSARevisionHistoryDeployment, "", "kubectl", "apply", "-n", DeploymentNamespace(), "--server-side=true", "--field-manager=revision-history-manager", "--validate=false", "--force-conflicts", "-f", "-")
+			assert.NoError(t, err)
+			assert.Contains(t, output, "serverside-applied")
+		}).
+		Refresh(RefreshTypeNormal).
+		Then().
+		Expect(SyncStatusIs(SyncStatusCodeOutOfSync)).
+		Given().
+		ResourceOverrides(map[string]ResourceOverride{"apps/Deployment": {
+			IgnoreDifferences: OverrideIgnoreDiff{
+				ManagedFieldsManagers: []string{"revision-history-manager"},
+				JSONPointers:          []string{"/spec/template/spec/containers/0/image"},
+			},
+		}}).
+		When().
+		Refresh(RefreshTypeNormal).
+		Then().
+		Expect(SyncStatusIs(SyncStatusCodeSynced)).
+		Given().
+		When().
+		Sync().
```

Comment:
I think app should be patched before syncing, right? Can you swap these two lines?

Reply:
Actually this is to test the old behaviour before enabling the `RespectIgnoreDifferences=true`.